web/src/modules/i18n/BrowserLanguageDetector.js

```diff
@@ -0,0 +1,65 @@
+let hasLocalStorageSupport = null
+
+const localStorageAvailable = () => {
+  if (hasLocalStorageSupport !== null) {
+    return hasLocalStorageSupport
+  }
+
+  try {
+    hasLocalStorageSupport = window !== 'undefined' && window.localStorage !== null
+    const testKey = 'i18next.translate.boo'
+    window.localStorage.setItem(testKey, 'foo')
+    window.localStorage.removeItem(testKey)
+  } catch (e) {
+    hasLocalStorageSupport = false
+  }
+  return hasLocalStorageSupport
+}
+
+const LANGUAGE_LOCAL_STORAGE = 'i18nextLng'
+
+export default {
+  type: 'languageDetector',
+  async: false,
+  init: (services, detectorOptions, i18nextOptions) => {},
+  // https://github.com/i18next/i18next-browser-languageDetector/blob/master/src/browserLookups/navigator.js
+  // Returns array of ISO-639-2 or ISO-639-3 language codes
+  detect: () => {
+    const bcp47Tags = []
+
+    if (localStorageAvailable()) {
+      const localStorageLanguage = window.localStorage.getItem(LANGUAGE_LOCAL_STORAGE)
+      if (localStorageLanguage) {
+        bcp47Tags.push(localStorageLanguage)
+      }
+    }
+
+    if (typeof navigator !== 'undefined') {
+      if (navigator.languages) { // chrome only; not an array, so can't use .push.apply instead of iterating
```

Comment:
Why not just use `apply` then?

Reply:
what do you mean with apply?

I will not change this code as it is tested and works in all browsers. I will add tests though.

Reply:
I don't understand the comment: `// chrome only; not an array, so can't use .push.apply instead of iterating`. Or rather, why there is a comment instead of just applying the suggestion from the comment.

Reply:
This is from the upstream repoistory, not my comment :D

Reply:
I think we should decide if this is our own code now or just copied. If it is just copied, add a reference. If it is our own code it should also look like our own code :stuck_out_tongue: